src/app/mycart/over-all-total.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'overAllTotal',
+  pure:false
+})
+export class OverAllTotalPipe implements PipeTransform {
+
+  transform(value:any,products:any){
+    return this.overAllTotal(value,products);
+  }
+
+  overAllTotal(sum,products){
+    sum=products.reduce((acc,item)=>acc+(item.quantity*item.price),0);
```

Comment:
what is the purpose of passing sum as an argument if you are going to overwrite the value?